# Regex Basics Assignment

**Question 1** Count Phone Numbers in Text

**Description:** Write a Python function that counts how many phone numbers are present in a given long text. The phone numbers are in a 15-digit format.


In [ ]:
import re

def find_total_phone_numbers_present_in_text(sentence):
  return len(re.findall(r'\d{15}', sentence))
  # return len(re.findall(r'\d\d\d\d\d\d\d\d\d\d\d\d\d\d\d', sentence))


# main() code
text = """
Lorem ipsum dolor sit amet, 123456789012345 consectetur adipiscing elit.
Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
ad minim veniam, quis nostrud exercitation 987654321098765 ullamco laboris
nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit
in voluptate velit esse cillum dolore eu fugiat nulla pariatur. 555555555555555
Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt
mollit anim id est laborum. 111222333444555 000999888777666
"""

print(f"There are total {find_total_phone_numbers_present_in_text(text)} phone numbers (15 - digit format) present in above text!")

There are total 5 phone numbers (15 - digit format) present in above text!


**Question 2** Find Occurrences of a String in Text

**Description:** Write a Python function that checks how many times a particular substring is present in a given large text. Additionally, find the start index of the 3rd match and the end index of the 7th match. Return the sum of these indices.

In [ ]:
import re

def find_count_of_substring_presents_in_text(sentence, sub_sentence):
  print(f"Total {len(re.findall(sub_sentence, sentence))} times {sub_sentence} present in above text!")

  counter = 0
  sum_of_3rd_start_and_7th_end_indices = 0
  for sub_text in re.finditer(sub_sentence, sentence):
    counter += 1
    if(counter == 3):
      print(f"Starting index of {sub_sentence} at 3rd match is {sub_text.start()}")
      sum_of_3rd_start_and_7th_end_indices += sub_text.start()

    if (counter == 7):
      print(f"Ending index of {sub_sentence} at 7th match is {sub_text.end()}")
      sum_of_3rd_start_and_7th_end_indices += sub_text.end()
      break

  print(f"Sum of both indices (starting 3rd match and ending 7th match) is {sum_of_3rd_start_and_7th_end_indices}")



# main() code
text = """
The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.
The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.
The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.
The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.
"""

substring = "fox"
find_count_of_substring_presents_in_text(text, substring)

Total 8 times fox present in above text!
Starting index of fox at 3rd match is 108
Ending index of fox at 7th match is 293
Sum of both indices (starting 3rd match and ending 7th match) is 401


**Answer**

Count: 8

Start index of 3rd match: 108

End index of 7th match: 293

Sum of indices: 401